src/app/betting/feature/place-bet-page.component.ts

```diff
@@ -78,112 +80,115 @@ import { toSignal } from '@angular/core/rxjs-interop';
-          optionLabel="ssEventName"
-          styleClass="flex"
-          emptyMessage="No open special space events"
-          placeholder="Select a special space event"
-        />
-      </label>
-
-      <!-- Over/Under Radio Buttons -->
-      <div class="flex flex-wrap gap-3 mt-5">
-        <div class="flex align-items-center">
-          <p-radioButton
-            name="overUnder"
-            value="Over"
-            inputId="over"
-            [(ngModel)]="selectedOuOption"
-            styleClass="flex"
+    @switch (selectedBetType()) {
+      @case (BetType.DuelWinner) {
+        <!-- Duel Dropdown -->
+        <label class="flex flex-column gap-2 mt-5">
+          Duel
+          <p-dropdown
+            [options]="openDuels()"
+            [(ngModel)]="selectedDuel"
+            optionLabel="duelName"
+            styleClass="w-full"
+            emptyMessage="No open duels"
+            placeholder="Select a duel"
           />
-          <label for="over" class="ml-2"> Over </label>
-        </div>
-        <div class="flex align-items-center">
-          <p-radioButton
-            name="overUnder"
-            value="Under"
-            inputId="under"
-            [(ngModel)]="selectedOuOption"
-            styleClass="flex"
+        </label>
+
+        <!-- Duel Winner Dropdown -->
+        <label class="flex flex-column gap-2 mt-5">
+          Winner
+          <p-dropdown
+            [options]="competitors()"
+            [(ngModel)]="selectedWinner"
+            optionLabel="display_name"
+            styleClass="w-full"
+            placeholder="Select a winner"
           />
-          <label for="under" class="ml-2"> Under </label>
+        </label>
+      }
+      @case (BetType.SpecialSpaceEvent) {
+        <!-- SS Event Dropdown -->
+        <label class="flex flex-column gap-2 mt-5">
+          Special Space Event
+          <p-dropdown
+            [options]="openSsEvents()"
+            [(ngModel)]="selectedSsEvent"
+            optionLabel="ssEventName"
+            styleClass="w-full"
+            emptyMessage="No open special space events"
+            placeholder="Select a special space event"
+          />
+        </label>
+
+        <!-- Over/Under Radio Buttons -->
+        <div class="flex flex-wrap gap-3 mt-5">
+          <div class="flex align-items-center">
+            <p-radioButton
+              name="overUnder"
+              value="Over"
+              inputId="over"
+              [(ngModel)]="selectedOuOption"
+              styleClass="w-full"
+            />
+            <label for="over" class="ml-2"> Over </label>
+          </div>
+          <div class="flex align-items-center">
+            <p-radioButton
+              name="overUnder"
+              value="Under"
+              inputId="under"
+              [(ngModel)]="selectedOuOption"
+              styleClass="w-full"
+            />
+            <label for="under" class="ml-2"> Under </label>
```

Comment:
Nitpick (feel free to ignore)

If the `<p-radioButton>` is nested inside the `<label>`, we don't need to specify a `for` and `input` label. It's also less error-prone, since it removes the possibility of the 2 attributes mismatching.